src/EditorFeatures/Core/Implementation/Diagnostics/AbstractDiagnosticsTaggerProvider.AggregatingTagger.cs

```diff
@@ -74,6 +133,59 @@ private class AggregatingTagger : ForegroundThreadAffinitizedObject, IAccurateTa
                 _owner._diagnosticService.DiagnosticsUpdated += OnDiagnosticsUpdated;
             }
 
+            private void OnWorkspaceChanged(object sender, EventArgs e)
+            {
+                this.AssertIsForeground();
+
+                // Disconnect from the old workspace notifications and hook up to the new ones.
+                lock (_gate)
+                {
+                    if (_workspace != null)
```

Comment:
❔ Is it possible for `_workspace == _workspaceRegistration.Workspace` here?

Reply:
That shouldn't be possible, i don't think :-/

Reply:
I think there's some theoretical insanity where while we have queued this notification another thread goes off and changes it under you again. In any case, the pattern here is what I would recommend.